### PR TITLE
Makes being thrown into airlocks/doors/machines hurt, mobs can now be thrown over railings.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Title. You can now red grab mobs and throw them over railings for cool effect!

You can also be thrown over railings by natural causes, like the Leviathan firing. Be careful...

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -526,3 +526,12 @@ Class Procs:
...
 	return
+
+/obj/machinery/hitby(atom/movable/AM, var/speed = THROWFORCE_SPEED_DIVISOR)
+	. = ..()
+	if(isliving(AM))
+		var/mob/living/M = AM
+		M.turf_collision(src, speed)
+		return
+	else
+		visible_message("<span class='danger'>[src.name] was hit by [AM].</span>")
```
Comment (SleepyGemmy) [inline code/game/machinery/machinery.dm:537]:
```suggestion
		visible_message(SPAN_DANGER("\The [src.name] was hit by \the [AM]."))
```

File: code/game/objects/structures/window.dm [hunk 2]
```diff
@@ -171,16 +171,23 @@
...
 	..()
-	visible_message(SPAN_DANGER("[src] was hit by [AM]."))
 	var/tforce = 0
 	if(ismob(AM))
+		if(isliving(AM))
+			var/mob/living/M = AM
+			M.turf_collision(src, speed, /decl/sound_category/glasscrack_sound)
+			return
+		else
+			visible_message(SPAN_DANGER("[src] was hit by [AM]."))
```
Comment (SleepyGemmy) [inline code/game/objects/structures/window.dm:183]:
```suggestion
			visible_message(SPAN_DANGER("\The [src] was hit by \the [AM]."))
```

File: code/game/objects/structures/window.dm [hunk 3]
```diff
@@ -171,16 +171,23 @@
...
 		tforce = 40
 	else if(isobj(AM))
+		visible_message(SPAN_DANGER("[src] was hit by [AM]."))
```
Comment (SleepyGemmy) [inline code/game/objects/structures/window.dm:186]:
```suggestion
		visible_message(SPAN_DANGER("\The [src] was hit by \the [AM]."))
```

File: code/game/machinery/machinery.dm [hunk 4]
```diff
@@ -526,3 +526,12 @@ Class Procs:
...
 	return
+
+/obj/machinery/hitby(atom/movable/AM, var/speed = THROWFORCE_SPEED_DIVISOR)
+	. = ..()
+	if(isliving(AM))
+		var/mob/living/M = AM
+		M.turf_collision(src, speed)
+		return
+	else
+		visible_message(SPAN_DANGER("\The [src.name] was hit by \the [AM]."))
```
Comment (SleepyGemmy) [inline code/game/machinery/machinery.dm:537]: out of curiosity, is `src.name` necessary here instead of just `src`?

Comment (NonQueueingMatt) [reply code/game/machinery/machinery.dm:537]: Not really lmao it's oldcode but it's the same thing

Comment (SleepyGemmy) [inline code/game/machinery/machinery.dm:537]:
```suggestion
		visible_message(SPAN_DANGER("\The [src] was hit by \the [AM]."))
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**